### PR TITLE
Google Opitmize bugFix:  Adding script inside head if head is available

### PR DESCRIPTION
**Pull request by abandonned**
## Description of the change

Adding script inside head if head is available

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/425)

### CODE REVIEW

File: integrations/ScriptLoader.js [hunk 1]
```diff
@@ -9,10 +10,18 @@ const ScriptLoader = (id, src, async = defaultAsyncState) => {
   js.async = async === undefined ? defaultAsyncState : async;
   js.type = "text/javascript";
   js.id = id;
-  const e = document.getElementsByTagName("script")[0];
-  logger.debug("==parent script==", e);
-  logger.debug("==adding script==", js);
-  e.parentNode.insertBefore(js, e);
+  const h = document.getElementsByTagName("head")[0];
```
Comment (saikumarrs) [inline integrations/ScriptLoader.js:13]:
```suggestion
  const headElems = document.getElementsByTagName("head");
```

File: integrations/ScriptLoader.js [hunk 2]
```diff
@@ -9,10 +10,18 @@ const ScriptLoader = (id, src, async = defaultAsyncState) => {
...
+  if (Object.keys(h).length !== 0) {
+    const e = h.getElementsByTagName("script")[0];
```
Comment (saikumarrs) [inline integrations/ScriptLoader.js:15]: Can we not insert it before the first child of the head element? Does it have to be a script element?

Comment (abandonned) [reply integrations/ScriptLoader.js]: Updated.

File: integrations/ScriptLoader.js (hunk 1 above)
Comment (abandonned) [reply integrations/ScriptLoader.js]: Updated.

File: integrations/ScriptLoader.js [hunk 3]
```diff
@@ -9,10 +10,16 @@ const ScriptLoader = (id, src, async = defaultAsyncState) => {
...
   js.type = "text/javascript";
   js.id = id;
-  const e = document.getElementsByTagName("script")[0];
-  logger.debug("==parent script==", e);
-  logger.debug("==adding script==", js);
-  e.parentNode.insertBefore(js, e);
+  const headElems = document.getElementsByTagName("head");
+  if (Object.keys(headElems).length !== 0) {
+    logger.debug("==adding script==", js);
+    headElems.appendChild(js);
```
Comment (saikumarrs) [inline integrations/ScriptLoader.js:16]: We should not append it but rather insert it before the first child (it needn't be a script element)

Comment (abandonned) [reply integrations/ScriptLoader.js]: updated. Please take a look if its okay.

File: integrations/ScriptLoader.js [hunk 4]
```diff
@@ -9,10 +10,16 @@ const ScriptLoader = (id, src, async = defaultAsyncState) => {
...
   js.type = "text/javascript";
   js.id = id;
-  const e = document.getElementsByTagName("script")[0];
-  logger.debug("==parent script==", e);
-  logger.debug("==adding script==", js);
-  e.parentNode.insertBefore(js, e);
+  const headElems = document.getElementsByTagName("head");
+  if (Object.keys(headElems).length !== 0) {
+    logger.debug("==adding script==", js);
+    headElems.insertBefore(js, headElems.firstChild);
```
Comment (saikumarrs) [inline integrations/ScriptLoader.js:16]: Would this work if there is no first child?

Comment (abandonned) [reply integrations/ScriptLoader.js:16]: yes. should work even if firstChild is empty.

Comment (abandonned) [reply integrations/ScriptLoader.js:16]: https://www.w3.org/TR/DOM-Level-3-Core/core.html#ID-184E7107

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**